numpy/ma/tests/test_core.py

```diff
@@ -1485,6 +1485,11 @@ def test_eq_different_dimensions(self):
             assert_equal(test.mask, [[False, True],
                                      [False, True]])
 
+    def test_eq_optinfo(self):
+        a = array([1,2,2,4])
+        a._optinfo["key"] = "value"
+        assert_equal(a._optinfo["key"], (a == 2)._optinfo["key"])
```

Comment:
Can you extend this test to a bunch more operations? Arithmetic and comparison would be good.